client/state/themes/actions.js

```diff
@@ -454,3 +460,108 @@ export function clearThemeUpload( siteId ) {
 		siteId,
 	};
 }
+
+/**
+ * Start an Automated Transfer with an uploaded theme.
+ *
+ * @param {Number} siteId -- the site to transfer
+ * @param {File} file -- theme zip to upload
+ *
+ * @returns {Promise} for testing purposes only
+ */
+export function initiateThemeTransfer( siteId, file ) {
+	return dispatch => {
+		dispatch( {
+			type: THEME_TRANSFER_INITIATE_REQUEST,
+			siteId,
+		} );
+		return wpcom.undocumented().initiateTransfer( siteId, null, file, ( event ) => {
+			dispatch( {
+				type: THEME_TRANSFER_INITIATE_PROGRESS,
+				siteId,
+				loaded: event.loaded,
+				total: event.total,
+			} );
+		} )
+			.then( ( { transfer_id } ) => {
+				dispatch( {
+					type: THEME_TRANSFER_INITIATE_SUCCESS,
+					siteId,
+					transferId: transfer_id,
+				} );
+				dispatch( pollThemeTransferStatus( siteId, transfer_id ) );
+			} )
+			.catch( error => {
+				dispatch( {
+					type: THEME_TRANSFER_INITIATE_FAILURE,
+					siteId,
+					error,
+				} );
+			} );
+	};
+}
+
+// receive a transfer status
+function transferStatus( siteId, transferId, status, message, themeId ) {
```

Comment:
could even call them `receiveTransferStatus` (in analogy to `receiveTheme` which looks similar)